pyteamcity/future/artifact.py

```diff
@@ -0,0 +1,93 @@
+import fnmatch
+import os
+
+from .core.utils import parse_date_string
+
+
+class Artifact(object):
+    def __init__(self, build, path=''):
+        self.build = build
+        self.path = path
+        teamcity = self.build.build_query_set.teamcity
+        url = self.build.api_url + '/artifacts/metadata/' + self.path
+        res = teamcity.session.get(url)
+        res.raise_for_status()
+        self._data = res.json()
+
+    @property
+    def name(self):
+        return self._data['name']
+
+    def getsize(self):
+        return self._data.get('size')
+
+    @property
+    def size(self):
+        return self.getsize()
+
+    @property
+    def modification_time(self):
+        return parse_date_string(self._data['modificationTime'])
+
+    def splitext(self):
+        return os.path.splitext(self.name)
+
+    @property
+    def ext(self):
+        return self.splitext()[1]
+
+    def fnmatch(self, pattern):
+        return fnmatch.fnmatch(self.name, pattern)
```

Comment:
I'm basically wondering if we could have something like an `ArtifactFinder` class that would hide this detail and do:

```python
af = ArtifactFinder(...)
artifacts = af.find("*.xml")  # returns collection of artifacts
```

... so you don't have to expose `Artifact.fnmarch()`, the artifact finder could just run `fnmatch(Artifact.name, "*.xml")`.